test/object_store/run

```diff
@@ -12,39 +12,22 @@ import shutil
 import signal
 import yaml
 from contextlib import contextmanager
+from dataclasses import dataclass
 
-print('Scylla is: ' + run.find_scylla() + '.')
 success = True
 
 ssl = '--ssl' in sys.argv
-if ssl:
-    cmd = run.run_scylla_ssl_cql_cmd
-    check_cql = run.check_ssl_cql
-else:
-    cmd = run.run_scylla_cmd
-    check_cql = run.check_cql
 
 test_tempdir = run.pid_to_dir(os.getpid())
 os.mkdir(test_tempdir)
-s3_server_address = os.environ['S3_SERVER_ADDRESS_FOR_TEST']
-s3_server_port = int(os.environ['S3_SERVER_PORT_FOR_TEST'])
-s3_public_bucket = os.environ['S3_PUBLIC_BUCKET_FOR_TEST']
-
-
-with open(test_tempdir + '/object_storage.yaml', 'w') as config_file:
-    yaml.dump({ 'endpoints': [
-            {
-                'name': s3_server_address,
-                'port': s3_server_port,
-            }
-        ]
-    }, config_file)
-
-def run_scylla_cmd(pid, dir):
-    global cmd
-    (c, e) = cmd(pid, dir)
-    c += ['--object-storage-config-file', test_tempdir + '/object_storage.yaml']
-    return (c, e)
+
+
+@dataclass
+class S3_Server:
```

Comment:
It very likely makes sense to split this patch int two as per it's described in the commit message for the ease of review

Reply:
i think you are referencing the commit message of 

>     * define a dedicated S3_server class which duck types MinioServer.
>       it will be used to represent S3 server in place of MinioServer if
>       S3 is used for testing
>     * prepare object_storage.yaml in get_scylla_with_s3(), so it is more
>       clear that we are using the same set of settings for launching
>       scylla

the 2nd item depends on the first. because i use `S3_server` as a parameter new function. but if i extract the first commit out, it does not make sense anymore. as it won't have any user . if we extract the 2nd item as the first commit, the new method's signature would be ugly. i stared at this commit for a while, but still cannot think of a reasonable solution to improve the reviewability..